core/drivers/crypto/caam/hal/imx_6_7/sub.mk

```diff
@@ -2,7 +2,10 @@ incdirs-y += ../common
 incdirs-y += ../../include
 incdirs-y += .
 
-srcs-y += hal_clk.c
+ifneq (, $(filter y, $(CFG_MX6) $(CFG_MX6UL)))
```

Comment:
`srcs-$(CFG_MX6) += hal_clk_mx6_mx6ul.c` seems enough, all mx6* flavours are `CFG_MX6=y`

Reply:
correct. And so I'll rename the file to hal_clk_mx6.c